File: 2020/golang/day1.go
package main

import "log"

func day1() {

	data := StringSliceToIntegerSlice(getInputData(2020, 1))
	day1part1(data)
	day1part2(data)
}

func day1part1(data []int) {

	for _, i := range data {
		for _, j := range data {
			if i+j == 2020 {
				log.Println("Answer for Day 1, Part is", i*j)
				return
			}
		}
	}

}

func day1part2(data []int) {

	for _, i := range data {
		for _, j := range data {
			for _, k := range data {
				if i+j+k == 2020 {
					log.Println("Answer for Day 1, Part is", i*j*k)
					return
				}
			}
		}
	}

}

File: 2022/python/day1.py
from utils import getInputData

data = getInputData(year=2022, day=1)
data.append('')

# Part 1

elf_data = []
elf_items = []
for item in data:
    if item == '':
        elf_data.append(elf_items)
        elf_items = []
        continue
    elf_items.append(int(item))

# print(elf_data)

elf_sums = [sum(calories) for calories in elf_data]

print("Answer for part 1 is", max(elf_sums))

elf_sums.sort(reverse=True)

print("Answer for part 2 is", sum(elf_sums[0:3]))
